framework/src/service/ServiceRegistrationBase.cpp

```diff
@@ -273,10 +274,11 @@ void ServiceRegistrationBase::Unregister()
         std::string message(
           "ServiceFactory UngetService implementation threw an exception");
         d->bundle->coreCtx->listeners.SendFrameworkEvent(
-          FrameworkEvent(FrameworkEvent::Type::FRAMEWORK_WARNING,
+          FrameworkEvent(FrameworkEvent::Type::FRAMEWORK_ERROR,
                          MakeBundle(d->bundle->shared_from_this()),
                          message,
-                         std::current_exception()));
+                         std::make_exception_ptr(ServiceException(message, 
```

Comment:
Since an exception was thrown, we want to preserve the exception message and use that string as the ```message``` in the ```ServiceException``` object. Otherwise we are losing information which may be useful to users.